### PR TITLE
[RLlib] Revert 41c9ef70.

### PR DESCRIPTION
**Pull request by gjoliver**
## Why are these changes needed?

Also:
Add validation to make sure multi-gpu and micro-batch is not used together.
Update A2C learning test to hit the microbatching branch.
Minor comment updates.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [*] Unit tests
   - [*] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by avnishn — APPROVED** [review]

File: release/rllib_tests/learning_tests/yaml_files/a2c/a2c-breakoutnoframeskip-v4.yaml [hunk 1]
```diff
@@ -17,3 +17,5 @@ a2c-breakoutnoframeskip-v4:
             [0, 0.0007],
             [20000000, 0.000000000001],
         ]
+        train_batch_size: 256
+        microbatch_size: 64
```
Comment (kouroshHakha) [inline release/rllib_tests/learning_tests/yaml_files/a2c/a2c-breakoutnoframeskip-v4.yaml:21]: microbatch_size: 256 works for sure if this didn't work.

File: rllib/algorithms/a2c/a2c.py [hunk 2]
```diff
@@ -128,24 +128,34 @@ def validate_config(self, config: AlgorithmConfigDict) -> None:
                     "Otherwise, microbatches of desired size won't be achievable."
                 )
 
+            if config["num_gpus"] > 1:
```
Comment (kouroshHakha) [inline rllib/algorithms/a2c/a2c.py:131]: nice.

**Review by kouroshHakha — APPROVED** [review]
why are we introducing fake_gpu tests again? (stamped)

File: release/rllib_tests/learning_tests/yaml_files/a2c/a2c-breakoutnoframeskip-v4.yaml (hunk 1 above)
Comment (gjoliver) [reply release/rllib_tests/learning_tests/yaml_files/a2c/a2c-breakoutnoframeskip-v4.yaml:21]: 👌

**Comment by gjoliver** [comment]
> why are we introducing fake_gpu tests again? (stamped)

It was deleted with last change.
Need to stick to the revert.

**Comment by gjoliver** [comment]
release tests here: https://buildkite.com/ray-project/release-tests-pr/builds/11763#_
multigpu ones are fixed. TF leaning test also works. Torch seems to suffer the same CUDA speed issue.
we should merge this and get release unblocked.

**Pull request merged**